File: API/_archive/Protobuf/python/usecase3_file_storage.py
import os
# import models_pb2

class MockUser:
    def __init__(self):
        self.name = "Alice"
    def ParseFromString(self, data):
        pass

def read_from_file():
    if not os.path.exists("users.bin"):
        print("users.bin not found. Writing dummy file.")
        with open("users.bin", "wb") as f:
            f.write(b"mock binary data")

    with open("users.bin", "rb") as f:
        data = f.read()
        # user = models_pb2.User()
        user = MockUser()
        user.ParseFromString(data)
        print(f"Read user from binary file: {user.name}")

if __name__ == "__main__":
    read_from_file()

File: API/_archive/Protobuf/golang/usecase3_file_storage.go
package main

import (
	"log"
	"os"
	// "google.golang.org/protobuf/proto"
)

type User struct { Id int32; Name string; Email string }

func writeToFile(user *User) {
	// data, _ := proto.Marshal(user)
	data := []byte("mock binary payload")
	
	f, err := os.OpenFile("users.bin", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()
	
	f.Write(data)
	log.Println("Appended protobuf payload to disk")
}

func main() {}
